pkg/log/rate_limited_logger.go

```diff
@@ -0,0 +1,89 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package log
+
+import (
+	"time"
+
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+)
+
+type rateLimitedLogger struct {
+	logger  *zap.Logger
+	limiter *rate.Limiter
+}
+
+// RateLimitedLogger provides a logger that only logs once within a specified duration
+func RateLimitedLog(handle *LoggerHandle, every time.Duration) *rateLimitedLogger {
+	return &rateLimitedLogger{
+		logger:  Log(handle),
+		limiter: rate.NewLimiter(rate.Every(every), 1),
+	}
+}
+
+// visible for testing
+func TestingRateLimitedLog(logger *zap.Logger, every time.Duration) *rateLimitedLogger {
```

Comment:
Nit:
* rename to sth like `getTestRateLimitedLog()`
* move it to `rate_limited_logger_test.go`

Reply:
Many thanks, addressed comments in [0376571](https://github.com/apache/yunikorn-core/pull/663/commits/0376571ce3f0d40c0f7f76cb6a9bf8bfad66f57e)